home-work-app/src/app/redux/reducers/fetchMovies.ts

```diff
@@ -0,0 +1,26 @@
+import { IApiResponse } from '../../interfaces/IApiResponse';
+
+type FetchState = {
+    currentMovies: IApiResponse.IMovie[]
+};
+
+type FetchAction = {
+    type: string,
+    payload: IApiResponse.IMovie[]
+};
+type Reducer = (
+    state: FetchState,
+    action: FetchAction
+) => FetchState;
+
+const initialFetchState: FetchState = {
+    currentMovies: [],
+};
+
+export const fetchMovies: Reducer = (state: FetchState = initialFetchState, action: FetchAction) => {
```

Comment:
This `reducer` is incorrect. Reducer shouldn't mutate the state, it return new state every time action is passed to `reducer.`
It is one of main `redux` principles.
[https://redux.js.org/understanding/thinking-in-redux/three-principles](url)